resources/manage.rb

```diff
@@ -99,55 +102,55 @@
       end
 
       # use the keyfile defined in the databag or fallback to the standard file in the home dir
-      key_file = u['authorized_keys_file'] || "#{home_dir}/.ssh/authorized_keys"
+      key_file = user[:authorized_keys_file] || "#{home_dir}/.ssh/authorized_keys"
 
       template key_file do
         source 'authorized_keys.erb'
         cookbook new_resource.cookbook
-        owner u['uid'] ? validate_id(u['uid']) : u['username']
-        group validate_id(u['gid']) if u['gid']
+        owner user[:uid] ? validate_id(user[:uid]) : username
+        group validate_id(user[:gid]) if user[:gid]
         mode '0600'
         sensitive true
-        # ssh_keys should be a combination of u['ssh_keys'] and any keys
+        # ssh_keys should be a combination of user['ssh_keys'] and any keys
         # returned from a specified URL
         variables ssh_keys: ssh_keys
-        only_if { !!(u['ssh_keys']) }
+        only_if { !!(user[:ssh_keys]) }
       end
 
-      if u['ssh_private_key']
-        key_type = u['ssh_private_key'].include?('BEGIN RSA PRIVATE KEY') ? 'rsa' : 'dsa'
+      if user[:ssh_private_key]
+        key_type = user[:ssh_private_key].include?('BEGIN RSA PRIVATE KEY') ? 'rsa' : 'dsa'
```

Comment:
I'd move this into a helper method so it'll be easier to add support for ECDSA and ED25519 etc